lib/elements/ordering_buffer.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Basic.Elements.OrderingBuffer do
+  @moduledoc """
+  An element that gathers the packets and puts them in the order, sorted by their sequence id.
+  Once the consistent batch of packets (which means - with no packets missing in-between) is completely gathered, it is sent through the output pad.
+  """
+  use Membrane.Filter
+  alias Basic.Formats.Packet
+
+  def_input_pad :input,
+                [
+                  demand_unit: :buffers,
+                  accepted_format: %Packet{type: :custom_packets}
+                ]
+
+  def_output_pad :output,
+                 [
+                   accepted_format: %Packet{type: :custom_packets}
+                 ]
+
+  @impl true
+  def handle_init(_context, _options) do
+    {[],
+      %{
+        ordered_packets: [],
+        last_sent_seq_id: 0
+      }
+    }
+  end
+
+  @impl true
+  def handle_demand(:output, size, _unit, _context, state) do
+    {[demand: {:input, size}], state}
+  end
+
+  @impl true
+  def handle_process(:input, buffer, _context, state) do
+    packet = unzip_packet(buffer.payload)
+    ordered_packets = [packet | state.ordered_packets] |> Enum.sort()
```

Comment:
[NIT] You might consider:
```suggestion
    ordered_packets = Enum.sort( [packet | state.ordered_packets] )
```